sql.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2014 Google Inc.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uuid
+
+import (
+	"database/sql"
+	"database/sql/driver"
```

Comment:
I would be fine with Scan, but I do not want to pull a dependency in on sql or sql driver.  That would force the inclusion of sql and sql/driver in every application that uses UUID's.  Seems to me the functionality should live in an sql library.  You can simmly have sql/uuid which has "type UUID struct { uuid.UUID } and then define your new types on (or I believe you can).